### ISSUE TITLE
Button- Accessibility violation- Nested interactive controls are not announced by screen readers

### ISSUE THREAD

**Issue by divyajindel** [Sage/carbon#4632]
### Current behaviour

The Button has an accessibility violation "Ensure interactive controls are not nested"
for example - https://carbon.sage.com/?path=/story/button--icon-only-button-with-tooltip

### Expected behaviour

There are no accessibility violations.

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/button--icon-only-button-with-tooltip

### Carbon Version

latest

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by samtjo** [comment]
FE-4630
